ConcurrentModificationException

Caused by: java.util.ConcurrentModificationException
	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:909) ~[?:1.8.0_212]
	at java.util.ArrayList$Itr.next(ArrayList.java:859) ~[?:1.8.0_212]
	at org.zeroBzeroT.antispam.SpamCheck.isSpam(SpamCheck.java:40) ~[?:?]
	at org.zeroBzeroT.antispam.AntiSpam.onPlayerCommandPreprocess(AntiSpam.java:134) ~[?:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor13.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:70) ~[patched_1.12.2.jar:git-Paper-1596]
	... 16 more

--- a/src/main/java/org/zeroBzeroT/antispam/SpamCheck.java
+++ b/src/main/java/org/zeroBzeroT/antispam/SpamCheck.java
@@ -3,6 +3,7 @@ package org.zeroBzeroT.antispam;
 import org.bukkit.entity.Player;
 
 import java.util.Arrays;
+import java.util.LinkedList;
 
 public class SpamCheck {
 	// factor of the message difference - TODO: Config vars
@@ -37,7 +38,7 @@ public class SpamCheck {
 		int cntDuplicates = 0;
 
 		// has the same already been written?
-		for (String oldMsg : lastMessages) {
+		for (String oldMsg : new LinkedList<>(lastMessages)) { // copy contents to new object to avoid concurrent modification by async chat event handling
 			// difference in length of the messages is already greater than the factor
 			if (Math.abs(oldMsg.length() - saniMsg.length()) > Math.max(oldMsg.length(), saniMsg.length()) * msgDiffFactor)
 				continue;

